cypress/integration/spec.js

```diff
@@ -76,31 +88,90 @@ describe('Chart Explorer', function() {
     ).click()
     cy.get('[data-test=rm-encoding]').click()
 
-    // check for editor existence
-    cy.get('#configure-tabs-tab-editor').click()
-    cy.get('.monaco-editor')
-    cy.get('#configure-tabs-tab-builder').click()
-
     // check that titles work
     cy.get('[data-test=chart-title]').type('test title')
     cy.get('svg .role-title').contains('test title')
 
     // check that chart exists
     cy.get('[data-test=vega-embed]')
-
     cy.screenshot('page with viz')
 
-    // check download options
+    cy.get('[data-test=license-open]').click()
+    cy.get('[data-test=license-text]').contains('cypress license text')
+  })
+
+  it('Should be able to edit chart using editor', function() {
+    cy.get('#configure-tabs-tab-editor').click()
+
+    cy.get('.inputarea')
+      .type('{ctrl}f')
+      .focused()
+      .type('"title": "test 123",')
+
+    cy.get('#configure-tabs-tab-builder').click()
+
+    cy.get('svg .role-title').contains('test 123')
+  })
+
+  it('Should be able to download a chart', function() {
     cy.get('#dropdown-download').click()
 
+    // download as a vega-lite file format
     cy.get('.open > .dropdown-menu > :nth-child(3) > a').trigger('mousedown')
     cy.get('.open > .dropdown-menu > :nth-child(3) > a').should($m => {
       expect($m).to.have.length(1)
       expect($m.attr('href')).to.match(/^blob/)
     })
-    cy.get('#dropdown-download').click()
 
-    // check download as insight
+    // download as a vega file format
+    cy.get('.open > .dropdown-menu > :nth-child(2) > a').trigger('mousedown')
+    cy.get('.open > .dropdown-menu > :nth-child(2) > a').should($m => {
+      expect($m).to.have.length(1)
+      expect($m.attr('href')).to.match(/^blob/)
+    })
+
+    // download as a png file format
+    cy.get('.open > .dropdown-menu > :nth-child(5) > a').trigger('mousedown')
+    cy.get('.open > .dropdown-menu > :nth-child(5) > a').should($m => {
+      expect($m).to.have.length(1)
+      expect($m.attr('href')).to.match(/^data:image\/png/)
+    })
+
+    // download as a svg file format
+    cy.get('.open > .dropdown-menu > :nth-child(6) > a').trigger('mousedown')
+    cy.get('.open > .dropdown-menu > :nth-child(6) > a').should($m => {
+      expect($m).to.have.length(1)
+      expect($m.attr('href')).to.match(/^blob/)
+    })
+
+    // download as a HTML file format
+    cy.get('.open > .dropdown-menu > :nth-child(8) > a').trigger('mousedown')
+    cy.get('.open > .dropdown-menu > :nth-child(8) > a').should($m => {
+      expect($m).to.have.length(1)
+      expect($m.attr('href')).to.match(/^blob/)
+    })
+  })
+
+  it('Should be able to be shared', function() {
+    cy.get('[data-test=chart-title]').type('test title')
+
+    // share as an insight
+    cy.get('#dropdown-save-ddw').click()
+    cy.get('.open > .dropdown-menu > :nth-child(2) > a').click()
+    cy.get('.modal')
+
+    cy.get('[type="checkbox"]').uncheck()
```

Comment:
this seems like too vague of a selector to me. is this to target a checkbox that is conveniently the only checkbox on the page? if so, if we ever added another checkbox to the doc, it seems like this would break pretty easily.